test/core/utils/filter-html-attrs.js

```diff
@@ -0,0 +1,49 @@
+describe('axe.utils.filterHtmlAttrs', function() {
+  'use strict';
+  var fixture = document.querySelector('#fixture');
+
+  var filterHtmlAttrs = axe.utils.filterHtmlAttrs;
+  var html, expected;
+
+  beforeEach(function() {
+    fixture.innerHTML =
+      '<label> My Label <input type="text" value="my value" /></label>';
+    html = fixture.firstChild;
+    expected = '<label> My Label <input value="my value"></label>';
+  });
+
+  it('should return string if no attributes are passed', function() {
+    assert.equal(filterHtmlAttrs(html), html);
+  });
+
+  it('should filter attribute if exists', function() {
+    assert.equal(filterHtmlAttrs(html, { type: true }).outerHTML, expected);
+  });
+
+  it('should filter attribute if matches CSS selector', function() {
+    assert.equal(filterHtmlAttrs(html, { type: 'input' }).outerHTML, expected);
```

Comment:
Can you put in an actual CSS string here? This could just be nodeName match still.

Reply:
This is a CSS string? I'm matching the `type` attribute only if the element is an `input` node.